wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -964,6 +968,51 @@ void downloadS3() throws ModelException, IOException {
         }
     }
 
+    void build_trt_llm_engine() throws ModelException, IOException {
+        logger.info("Start building TensorRT-LLM engine");
+        final String trt_llm_model_repo = "/tmp/tensorrtllm";
+        prop.putIfAbsent("option.trt_llm_model_repo", trt_llm_model_repo);
+        
+        // writing updated properties file after env variables are added
+        final Path propertiesDir = Paths.get("/tmp/properties");
+        final Path propertiesFile = propertiesDir.resolve("serving.properties");
+        if (Files.exists(propertiesFile)) {
+            Files.delete(propertiesFile);
+        }
+        Files.createDirectories(propertiesFile.getParent());
+        FileOutputStream fileOutputStream = new FileOutputStream(propertiesFile.toFile());
+        prop.store(fileOutputStream, "write serving.properties");
+        fileOutputStream.close();
+        
+        // invoke trt-llm build script
+        List<String> commandList = new ArrayList<>();
+        commandList.add("python");
+        commandList.add("/opt/djl/partition/trt_llm_partition.py");
+        commandList.add("--properties_dir");
+        commandList.add(propertiesDir.toString());
```

Comment:
Sometimes people pass in all properties through environment variables. So there is no physical  file

Reply:
Right, this is not relying on user set `serving.properties`. I am writing a new file in #L984 and by that time `prop` is [updated](https://github.com/deepjavalibrary/djl-serving/blob/master/wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java#L703-L710) with environment variables already, so it will env variables information as well.

Reply:
You can save the new information to PyEnv instead of writing to a file

Reply:
using existing serving.properties file and reading env vars on the python side.